packages/framework/src/Model/CombinedArticle/CombinedArticleElasticsearchRepository.php

```diff
@@ -54,17 +68,18 @@ protected function extractHits(array $result): array
 
     /**
      * @param string $indexVersion
+     * @param int|null $domainId
      * @return string
      */
-    protected function getIndexNameFromIndexVersion(string $indexVersion): string
+    protected function getIndexNameFromIndexVersion(string $indexVersion, ?int $domainId = null): string
```

Comment:
What about making the domainId mandatory here, in the `extractHits`, and in the `getArticlesBySearchText` methods? 
I see a few benefits:
- the repository will be much simpler
- usage will be the same for getArticles*() methods (currently one accepts domain, while the other doesn't and that may be confusing)
- We may then make the repository domain independent completely

Reply:
great idea, refactored in 8ab1187e1e